cli/azd/pkg/project/service_models.go

```diff
@@ -78,20 +78,38 @@ func (spr *ServicePackageResult) ToString(currentIndentation string) string {
 		return uxItem.ToString(currentIndentation)
 	}
 
-	return fmt.Sprintf("%s- Package Output: %s", currentIndentation, output.WithLinkFormat(spr.PackagePath))
+	return fmt.Sprintf("%s- Package Output: %s", currentIndentation, output.WithHighLightFormat(spr.PackagePath, ""))
 }
 
 func (spr *ServicePackageResult) MarshalJSON() ([]byte, error) {
 	return json.Marshal(*spr)
 }
 
+// Endpoint is a URL with an option name and description
+type Endpoint struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Url         string `json:"url"`
+}
+
+// String returns a string representation of the endpoint
+func (s *Endpoint) String() string {
+	link := output.WithHyperlink(s.Url, s.Name)
+
+	if s.Description != "" {
+		return fmt.Sprintf("Endpoint: %s %s", link, output.WithGrayFormat("(%s)", s.Description))
+	}
+
+	return fmt.Sprintf("Endpoint: %s", link)
+}
+
 // ServiceDeployResult is the result of a successful Deploy operation
 type ServiceDeployResult struct {
 	Package *ServicePackageResult `json:"package"`
 	// Related Azure resource ID
 	TargetResourceId string            `json:"targetResourceId"`
 	Kind             ServiceTargetKind `json:"kind"`
-	Endpoints        []string          `json:"endpoints"`
+	Endpoints        []*Endpoint       `json:"endpoints"`
```

Comment:
I would prefer value over pointer semantics here, given that there shouldn't be any shared modification. This should be a data struct. I wonder what others think about this.

Reply:
I don't have a strong preference here since we don't need to support modifying the data outside the functions.  I've just got into the habit of using pointers for the very small performance improvements instead of the data getting copied across function boundaries.

Reply:
In general, I'd much prefer value semantics by default, unless the intention is to have shared modification, dealing with large structs, or signifying optional behavior.

Value semantics may have comparable performance to pointer semantics due to stack allocation from compiler optimization or hardware caching. It's hard to imagine that this ever ends up being the bottleneck, which is why I'd prefer the method signature to signify the intended usage and slight readability improvements over the perf gains.

But here we have a pointer receiver `*String` and not using the returned value as a plain-old struct, so this is OK.